### PR TITLE
doc: readme improvements (cleaup, xcode debugging, etc)

### PR DESCRIPTION
**Pull request by ath0mas**
### Platforms affected
iOS

### Motivation and Context

Reviewing Readme between cordova-ios and cordova-android.

### Description

Fix typo and add "Debugging in Xcode".

(also see similar PR for cordova-android apache/cordova-android#1308)

### Testing

.

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change
- [x] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Comment by ath0mas** [comment]
I just applied the same changes suggested and merged by @erisu in linked PR for cdv-android.

Ready for review!

File: README.md [hunk 1]
```diff
@@ -47,20 +47,24 @@ cordova platform add --link /path/to/cordova-ios
 
 ## Updating a Cordova project
 
-When you install a new `cordova-cli` version that comes with a new iOS platform version, from within your project:
+When you install a new version of the [`Cordova CLI`](https://www.npmjs.com/package/cordova) that pins a new version of the [`Cordova-iOS`](https://www.npmjs.com/package/cordova-ios) platform, you can follow these simple upgrade steps within your project:
```
Comment (breautek) [inline README.md:50]: I think it's better to reference `Cordova-iOS` as the package name literally, `cordova-ios`, which also makes this consistent with other parts of the docs, such as:

> link the appropriate cordova-ios platform folder path:

```suggestion
When you install a new version of the [`Cordova CLI`](https://www.npmjs.com/package/cordova) that pins a new version of the [`cordova-ios`](https://www.npmjs.com/package/cordova-ios) platform, you can follow these simple upgrade steps within your project:
```

**Review by breautek — COMMENTED** [review]
All in all, I think these are good improvements. I Just have a couple of nit picks.

File: README.md [hunk 2]
```diff
@@ -7,9 +7,9 @@
 # to you under the Apache License, Version 2.0 (the
 # "License"); you may not use this file except in compliance
 # with the License.  You may obtain a copy of the License at
-# 
+#
```
[the remark this answers is not in the document]
Comment (ath0mas) [reply README.md]: Done

File: README.md [hunk 3]
```diff
@@ -7,9 +7,9 @@
...
 # http://www.apache.org/licenses/LICENSE-2.0
-# 
+#
```
[the remark this answers is not in the document]
Comment (ath0mas) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (ath0mas) [reply README.md:50]: This wording is to match with the code review changes done by @erisu in PR apache/cordova-android#1308 (see https://github.com/apache/cordova-android/pull/1308/commits/993737e0986f20893f37ebbdb864db0682ecbfc6).

Do you confirm I should reword it to `cordova-ios` here? and do it too for cordova-android through a new PR?

Comment (breautek) [reply README.md:50]: Ah well, I'm not really sure :sweat_smile:  and I can't really find an any document to point to that defines how things should be worded/referenced.

So we'll just say I'm wrong and we can follow the style as requested from Erisu.

**Review by breautek — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -47,20 +47,24 @@ cordova platform add --link /path/to/cordova-ios
...
+When you install a new version of the [`Cordova CLI`](https://www.npmjs.com/package/cordova) that pins a new version of the [`Cordova-iOS`](https://www.npmjs.com/package/cordova-ios) platform, you can follow these simple upgrade steps within your project:
 
 ```bash
 cordova platform rm ios
 cordova platform add ios
 ```
 
+## Debugging in Xcode
+
+Import project in Xcode through _File > Open_ and targeting `/path/to/your-cdv-project/platforms/ios/`.
```
Comment (NiklasMerz) [inline README.md:59]: You can also open the `PROJECTNAME.xcodeproj` or ``PROJECTNAME.xcodeworkspace` files in the platforms folder.

**Review by NiklasMerz — APPROVED** [review]

**Review by PieterVanPoyer — APPROVED** [review]

**Comment by erisu** [comment]
As for @NiklasMerz comment

> You can also open the PROJECTNAME.xcodeproj or ``PROJECTNAME.xcodeworkspace` files in the platforms folder.

IMO, if you want to add this to the README, only suggest opening the `xcodeworkspace`. Do not mention the option for `xcodeproj` or say that opening the `xcodeproj` will lead to issues.

If my memory is correct, opening the `xcodeproj` will not load the `CordovaLib` and leads to reference errors and build failures.

**Comment by NiklasMerz** [comment]
@erisu I agree. I have used both with no issues so far but we should only recommend the workspace.

**Comment by erisu** [comment]
OK, I have remembered it now and confirmed,

If a user is installing a plugin that uses Pods, opening the `xcodeproj` will not load the Pods as it is a separate project. Then, the code that tries to import the pods will fail and therefore the build fails.

If the project contains Pods, the users should open `xcworkspace`.

This is why I always suggested to people to open the `xcworkspace`. If we mentioned anything about `xcodeproj` including why not to open it because its nuances could lead to confusion.

If anything is mentioned, just the `xcworkspace` is fine.

Here is also an example screenshot of the XCode project navigator to show the difference.

| xcworkspace | xcodeproj |
| -- | -- |
| xcworkspace | xcodeproj |

File: README.md (hunk 4 above)
Comment (erisu) [reply README.md:59]:
```suggestion
Import project in Xcode through _File > Open_ and targeting `/path/to/your-cdv-project/platforms/ios/PROJECT_NAME.xcodeworkspace`.
```

File: README.md [hunk 5]
```diff
@@ -47,20 +47,24 @@ cordova platform add --link /path/to/cordova-ios
...
+When you install a new version of the [`Cordova CLI`](https://www.npmjs.com/package/cordova) that pins a new version of the [`Cordova-iOS`](https://www.npmjs.com/package/cordova-ios) platform, you can follow these simple upgrade steps within your project:
 
 ```bash
 cordova platform rm ios
 cordova platform add ios
 ```
 
+## Debugging in Xcode
+
+Import project in Xcode through _File > Open_ and targeting `/path/to/your-cdv-project/platforms/ios/PROJECT_NAME.xcodeworkspace`.
```
Comment (erisu) [inline README.md:59]:
```suggestion
Import project in Xcode through _File > Open_ and targeting `/path/to/your-cdv-project/platforms/ios/PROJECT_NAME.xcworkspace`.
```

**Review by erisu — APPROVED** [review]
